src/motors.c

```diff
@@ -29,7 +29,101 @@ static void init_motor_current_temp_MUX(){
 
 static void init_motor_speed_control()
 {
+	//Enable clock
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOA, ENABLE);
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOD, ENABLE);
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
+	//init timers
+	RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM2, ENABLE);
+	RCC_APB2PeriphClockCmd(RCC_APB2Periph_TIM1, ENABLE);
+	RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM4, ENABLE);
+	RCC_APB1PeriphClockCmd(RCC_APB1Periph_TIM3, ENABLE);
+	//GPIOA Configuration for motors 1-4 and 7
+	GPIO_InitTypeDef GPIOA_InitStruct;
+	GPIOA_InitStruct.GPIO_Pin = GPIO_Pin_0 | GPIO_Pin_3 | GPIO_Pin_10 | GPIO_Pin_8 | GPIO_Pin_6;
+	GPIOA_InitStruct.GPIO_Mode = GPIO_Mode_AF;
+	GPIOA_InitStruct.GPIO_Speed = GPIO_Speed_100MHz;
+	GPIOA_InitStruct.GPIO_OType = GPIO_OType_PP;
+	GPIOA_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(GPIOA, &GPIOA_InitStruct);
+	//GPIOD configuration for motors 5-6
+	GPIO_InitTypeDef GPIOD_InitStruct;
+	GPIOD_InitStruct.GPIO_Pin = GPIO_Pin_15 | GPIO_Pin_12;
+	GPIOD_InitStruct.GPIO_Mode = GPIO_Mode_AF;
+	GPIOD_InitStruct.GPIO_Speed = GPIO_Speed_100MHz;
+	GPIOD_InitStruct.GPIO_OType = GPIO_OType_PP;
+	GPIOD_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(GPIOD, &GPIOD_InitStruct);
+	//GPIOB configuration for motor 8
+	GPIO_InitTypeDef GPIOB_InitStruct;
+	GPIOB_InitStruct.GPIO_Pin = GPIO_Pin_0;
+	GPIOB_InitStruct.GPIO_Mode = GPIO_Mode_AF;
+	GPIOB_InitStruct.GPIO_Speed = GPIO_Speed_100MHz;
+	GPIOB_InitStruct.GPIO_OType = GPIO_OType_PP;
+	GPIOB_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
+	GPIO_Init(GPIOB, &GPIOB_InitStruct);
+	//pin alternate function timer configuration
+	GPIO_PinAFConfig(GPIOA, GPIO_PinSource0, GPIO_AF_TIM2);
+	GPIO_PinAFConfig(GPIOA, GPIO_PinSource3, GPIO_AF_TIM2);
+	GPIO_PinAFConfig(GPIOA, GPIO_PinSource10, GPIO_AF_TIM1);
+	GPIO_PinAFConfig(GPIOA, GPIO_PinSource8, GPIO_AF_TIM1);
+	GPIO_PinAFConfig(GPIOD, GPIO_PinSource15, GPIO_AF_TIM4);
+	GPIO_PinAFConfig(GPIOD, GPIO_PinSource12, GPIO_AF_TIM4);
+	GPIO_PinAFConfig(GPIOA, GPIO_PinSource6, GPIO_AF_TIM3);
+	GPIO_PinAFConfig(GPIOB, GPIO_PinSource0, GPIO_AF_TIM3);
+	//timer set up
+	TIM_TimeBaseInitTypeDef TIM_TimeBaseStructure;
+	TIM_TimeBaseStructure.TIM_Period = 1000 - 1;
```

Comment:
More resolution would be better. In reason we don't make this 10000 or 50000?

Reply:
How would I modify the period without affecting the value that will be inputted into the capture compare register?